alicloud/data_source_alicloud_db_instances.go

```diff
@@ -0,0 +1,261 @@
+package alicloud
+
+import (
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/rds"
+	"github.com/hashicorp/terraform/helper/schema"
+	"regexp"
+)
+
+func dataSourceAlicloudDBInstances() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAlicloudDBInstancesRead,
+
+		Schema: map[string]*schema.Schema{
+			"name_regex": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateNameRegex,
+			},
+			"engine": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validateAllowedStringValue([]string{
+					string(MySQL),
+					string(SQLServer),
+					string(PPAS),
+					string(PostgreSQL),
+				}),
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+				// please follow the link below to see more details on available statusesplease follow the link below to see more details on available statuses
+				// https://help.aliyun.com/document_detail/26315.html
+			},
+			"instance_type": {
```

Comment:
Maybe rds_typeo or category will be better here. Resource 'db_instance' use parameter 'instance_type' to specify its specification.

Reply:
aligned with resource `db_instance`